### PR TITLE
ARTEMIS-2210 Fix PagingStore creation synchronization issue 

### PR DESCRIPTION
**Pull request by gaohoward**
In PagingManagerImpl#getPageStore() the operations on the map 'stores'
are not synchronzed and it's possible that more than one paging store is
created for one address.

### CODE REVIEW

**Comment by gaohoward** [comment]
The fix is so obvious that I decide not to provide a test. :)

**Comment by franz1981** [comment]
@gaohoward The synchronized block will block any threads attempting to both getting and creating a paging store, while a chm::get is lock-free ie won't block any threads and chm::computeIfAbsent is locked on per-segment basis (that means that depends on the number of segments and how the hashing of the keys will distribute the entries into the segment.

Putting a synchronized like this will prevent any threads to do anything on an hot-path, blocking each others: my advice is to keep chm::get lock-free (as it is now) and into `newStore(storeName)` use chm::computeIfAbsent to avoid duplicates creatings of `PagingStore` instances.
This should keep the scalability as it is now without allocating unnecessary instances. wdyt?

**Comment by gaohoward** [comment]
@franz1981 ok, make sense. I'll change it. thx.

**Comment by gaohoward** [comment]
@franz1981 I've update the branch, can you take a look? thx.

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingManagerImpl.java [hunk 1]
```diff
@@ -335,19 +335,25 @@ public void deletePageStore(final SimpleString storeName) throws Exception {
...
    @Override
    public PagingStore getPageStore(final SimpleString storeName) throws Exception {
       if (managementAddress != null && storeName.startsWith(managementAddress)) {
          return null;
       }
-      PagingStore store = stores.get(storeName);
 
-      if (store != null) {
-         return store;
+      try {
```
Comment (franz1981) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingManagerImpl.java]: I think that `stores.get` is better that will be used in the fast path as it was before: `chm:.get` is completly lock-free, while `chm::computeIfAbsent` will lock on segment level ie better to fallback to `chm::computeIfAbsent` only if `chm::get` return `null`

Comment (gaohoward) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingManagerImpl.java]: ok got it. I've updated the branch. Please take a look again. thx.

**Comment by franz1981** [comment]
Looks fine to me buddy, just change the title, because no sync is happening anymore ;)

**Comment by gaohoward** [comment]
@franz1981 Title changed. Thanks mate!

**Pull request closed without merging**